quantum/process_keycode/process_swap_hands.c

```diff
@@ -0,0 +1,150 @@
+/* Copyright 2021 QMK
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "process_swap_hands.h"
+#include "quantum_keycodes.h"
+#include "action_tapping.h"
+#include "matrix.h"
+
+extern const keypos_t PROGMEM hand_swap_config[MATRIX_ROWS][MATRIX_COLS];
+
+static matrix_row_t swap_state[MATRIX_ROWS];
+
+static bool     swap_hands_enabled = false;
+static bool     oneshot_enabled    = false;
+static uint16_t oneshot_expiry     = 0;
+
+#if defined(ONESHOT_TIMEOUT) && (ONESHOT_TIMEOUT > 0)
+static inline bool has_oneshot_swaphands_timed_out(void) {
+    return (oneshot_expiry && timer_expired(timer_read(), oneshot_expiry));
+#endif
+}
+
+static inline void swap_hands_task(void) {
+#if defined(ONESHOT_TIMEOUT) && (ONESHOT_TIMEOUT > 0)
+    if (has_oneshot_swaphands_timed_out()) {
+        clear_oneshot_swaphands();
+    }
+#endif
+}
+
+void clear_oneshot_swaphands(void) {
+    oneshot_expiry     = 0;
+    swap_hands_enabled = false;
+    oneshot_enabled    = false;
+}
```

Comment:
`clear_oneshot_swaphands` is being called before the function is declared. 

```suggestion
void clear_oneshot_swaphands(void) {
    oneshot_expiry     = 0;
    swap_hands_enabled = false;
    oneshot_enabled    = false;
}

static inline void swap_hands_task(void) {
#if defined(ONESHOT_TIMEOUT) && (ONESHOT_TIMEOUT > 0)
    if (has_oneshot_swaphands_timed_out()) {
        clear_oneshot_swaphands();
    }
#endif
}

```